src/validation.h

```diff
@@ -829,6 +837,27 @@ class ChainstateManager
     //! Get all chainstates currently being used.
     std::vector<CChainState*> GetAll();
 
+    //! Construct and activate a Chainstate on the basis of UTXO snapshot data.
+    //!
+    //! Steps:
+    //!
+    //! - Initialize an unused CChainState.
+    //!
+    //! - Load its `CoinsViews` contents from `coins_file`.
+    //!
+    //! - Verify that the hash of the resulting coinsdb matches the expected hash
+    //!   per assumeutxo chain parameters.
+    //!
+    //! - Wait for our headers chain to include the base block of the snapshot.
+    //!
+    //! - "Fast forward" the tip of the new chainstate to the base of the snapshot,
+    //!   faking nTx* block index data along the way.
+    //!
+    //! - Move the new chainstate to `m_snapshot_chainstate` and make it our
+    //!   ChainstateActive().
+    NODISCARD bool ActivateSnapshot(
+        CAutoFile* coins_file, SnapshotMetadata metadata, bool in_memory);
```

Comment:
in commit bdf182e81e

this can't be nullptr, so `&` seems more appropriate. Also the metadata should be read-only here?

```suggestion
        CAutoFile& coins_file, const SnapshotMetadata& metadata, bool in_memory);
```

Reply:
Yup, good call. Done.